cmake/serial_manipulator_joint_types/serial_manipulator_joint_types.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+(C) Copyright 2011-2025 DQ Robotics Developers
+
+This file is part of DQ Robotics.
+
+    DQ Robotics is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    DQ Robotics is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with DQ Robotics.  If not, see <http://www.gnu.org/licenses/>.
+
+Contributors:
+- Juan Jose Quiroz Omana (juanjose.quirozomana@manchester.ac.uk)
+*/
+
+#include <iostream>
+#include <dqrobotics/DQ.h>
+#include <dqrobotics/robot_modeling/DQ_SerialManipulatorDH.h>
+#include <dqrobotics/robot_modeling/DQ_SerialManipulatorMDH.h>
+#include <dqrobotics/robot_modeling/DQ_SerialManipulatorDenso.h>
+
+
+using namespace DQ_robotics;
+
+/**
+ * @brief perform_tests This function tests the following methods: set_joint_type, set_joint_types,
+ *                      get_joint_type, and get_joint_types.
+ * @param robot  A serial manipulator robot.
+ * @param msg    A message to be displayed.
+ * @param is_denso Flag used to indicate if the robot is an object of the DQ_SerialManipulatorDenso class.
+ */
+template<typename T>
+void perform_tests(const std::shared_ptr<T>& robot,
+                   const std::string& msg,
+                   const bool& is_denso = false
+                   );
+
+
+int main()
+{
+    DQ_JointType R = DQ_JointType::REVOLUTE;
+
+    Matrix<double,5,7> dh_matrix(5,7);
+    dh_matrix <<11, 12, 13, 14, 15, 16, 17, // theta
+        21, 22, 23, 24, 25, 26, 27, // d
+        31, 32, 33, 34, 35, 36, 37, // a
+        41, 42, 43, 44, 45, 46, 47, // alpha
+        R,  R,  R,  R,  R,  R,  R;
+
+    auto dh_robot = std::make_shared<DQ_SerialManipulatorDH>(dh_matrix);
+    auto mdh_robot = std::make_shared<DQ_SerialManipulatorMDH>(dh_matrix);
+    auto denso_robot = std::make_shared<DQ_SerialManipulatorDenso>(MatrixXd::Ones(6,6));
+
+    perform_tests(dh_robot,  "DQ_SerialManipulatorDH");
+    perform_tests(mdh_robot, "DQ_SerialManipulatorMDH");
+    perform_tests(mdh_robot, "DQ_SerialManipulatorDenso", true);
+
+    return 0;
+}
+
+template<typename T>
+void perform_tests(const std::shared_ptr<T>& robot, const std::string& msg, const bool& is_denso){
```

Comment:
Hi @juanjqo.

This is a bit strange. You don’t need the flag if you’re using a template. You can check the type and switch accordingly. That will be more scalable.

Reply:
Hi @mmmarinho, thanks for your feedback. I removed the flag. Now, I am checking the type using the following approach 
```cpp
    auto denso = std::dynamic_pointer_cast<DQ_SerialManipulatorDenso>(robot);
    if (denso)
        target_joint_type = DQ_JointType::REVOLUTE;
```
I wonder if this is the right strategy.

Kind regards, 

Juancho

Reply:
Hi @juanjqo,

This would be a good example:
https://github.com/SmartArmStack/sas_common/blob/cc092fd6f156c4cabb6dd48d0476261c1a639067/include/sas_common/sas_common.hpp#L59

The thing to keep in mind with templates is to always try to solve things at compilation time. Your check is in runtime so it wouldn’t fully take advantage if the template capabilities. 

That example is C++17 compliant I believe. If we agree that it’s time to update our standard then it’s fine.

Otherwise you can use template specialization.

Reply:
Hi @juanjqo, I implemented the modifications. =)